### ISSUE TITLE
package is invalid: crx_signature_verification_initialization_failed

### ISSUE THREAD

**Issue by antoinegoutagny** [Constellation/crxmake#18]
I get an error "package is invalid: crx_signature_verification_initialization_failed" when trying to install a packaged app that has been generated using crxmake 2.2.0.

That must come from crxmake since manually packaging in Chrome (latest stable 31.0.1650.63, windows 8) with the same pem file does not raise the error.

The crx format must have changed, because the header looks like 4372 3234 0200 0000 3C01 0000 0001 0000 (hex), which according to the current doc (http://developer.chrome.com/extensions/crx.html) means a public key length of  15361 bytes, and a signature length of 1 byte, which does not make any sense.

I've also noticed that the pem file generated by chrome is now 1703 bytes long, when it used to be 916 bytes long. Could it be that crxmake does not handle the new pem file format?

**Comment by cielavenir** [comment]
Hi, did you install openssl_pkcs8 gem? For some time, Chrome denies PKCS1 pem.

cf. https://github.com/Constellation/crxmake/pull/13

**Comment by antoinegoutagny** [comment]
No, it isn't installed. Trying to do so now, but it fails with "openssl_pkcs8.c:3:25: fatal error: openssl/err.h: No such file or directory". That's another problem though.

**Comment by cielavenir** [comment]
Ah, well, Windows has some different situations...

**Comment by kiwistrongis** [comment]
Just encountered this bug, making crxmake unusable :(

Chrome Version 35.0.1916.153
Fedora 20 ( Linux 3.14.9-200 )

**Comment by cielavenir** [comment]
On Linux, `sudo gem install openssl_pkcs8` should be quite easy. Just try it.

**Comment by kiwistrongis** [comment]
nope failed. still getting the crx_signature_verification_initialization_failed error.

**Comment by icambron** [comment]
Me too.

OSX 10.10.2
Chrome 41.0.2272.104 (64-bit)

Installed the pkcs8 gem; that doesn't appear to be the problem. Using the Chrome UI to pack it works fine.

@antoinegoutagny I think you're misreading the output. It's all little endian, so what you're seeing is that the key is 316 and the sig is 256, which is how it's designed to work (316 = 294 byte key + 22 bytes tacked to the front). Now, whether that's _right_ is a separate question.

**Comment by icambron** [comment]
I submitted a fix for this as a PR, but I also created a new project to package extensions, and people might find it useful: https://github.com/icambron/extensionator

**Comment by kiwistrongis** [comment]
re:extensionator - thanks, i'll check it out soon as i'll probably need it
